### PR TITLE
fix: [PLG-3162]: fixing existing agent error

### PR DESCRIPTION
**Pull request by dhruvUpadhyay07**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Review by satyam-harness — APPROVED** [review]

File: src/modules/75-cd/pages/get-started-with-cd/CDOnboardingWizardWithCLI/Steps/DeploymentFlowTypes/CreateAgentModal.tsx [hunk 1]
```diff
@@ -96,10 +99,10 @@ export default function CreateGitopsAgentModal({
         onAgentCreated(res.status)
         setIsDownloadDisabed(false)
       })
-      .catch(e => {
-        if (shouldShowError(e)) {
-          showError(getErrorInfoFromErrorObject(e))
-        }
+      .catch(() => {
+        refetchHelmChart()
```
Comment (bhavya-sinha) [inline src/modules/75-cd/pages/get-started-with-cd/CDOnboardingWizardWithCLI/Steps/DeploymentFlowTypes/CreateAgentModal.tsx:103]: Is this expected . Calling API in catch?

Comment (dhruvUpadhyay07) [reply src/modules/75-cd/pages/get-started-with-cd/CDOnboardingWizardWithCLI/Steps/DeploymentFlowTypes/CreateAgentModal.tsx:103]: yes , we will be redesigning the flow , for now we are doing this

**Review by satyam-harness — DISMISSED** [review]

**Comment by dhruvUpadhyay07** [comment]
retrigger jest

File: src/modules/75-cd/pages/get-started-with-cd/CDOnboardingWizardWithCLI/Steps/DeploymentFlowTypes/CreateAgentModal.tsx [hunk 2]
```diff
@@ -96,10 +99,10 @@ export default function CreateGitopsAgentModal({
...
+        onAgentCreated(agent)
```
Comment (knagurski) [inline src/modules/75-cd/pages/get-started-with-cd/CDOnboardingWizardWithCLI/Steps/DeploymentFlowTypes/CreateAgentModal.tsx:104]: So if the call fails, you set a hard-coded agent and continue without notifying? Is that correct?

Comment (dhruvUpadhyay07) [reply src/modules/75-cd/pages/get-started-with-cd/CDOnboardingWizardWithCLI/Steps/DeploymentFlowTypes/CreateAgentModal.tsx:104]: yes

**Review by knagurski — DISMISSED** [review]

**Review by satyam-harness — DISMISSED** [review]

**Review by satyam-harness — APPROVED** [review]

**Review by batraprashant — DISMISSED** [review]

**Pull request merged**